### ISSUE TITLE
Tone.Players onload callback doesn't fire when there's more than one player 

### ISSUE THREAD

**Issue by ifjorissen** [Tonejs/Tone.js#1184]
**Describe the bug**
Players.onload callback doesn't seem to be invoked if the Players object is constructed with _more than one_ audio file / buffer.

**To Reproduce**

The code snippet below is taken from Players.test.ts. 

I imagine that if you added a second item to the url audio buffer map and ran the test, that the `onload` callback would not fire & the test would fail.

```
		it("can pass in additional args in the second parameters", (done) => {
			const players = new Players({
				test: "./audio/sine.wav",
			}, {
				onload: () => {
					expect(players.player("test").buffer.loaded).to.be.true;
					expect(players.volume.value).to.be.closeTo(-12, 0.1);
					players.dispose();
					done();
				},
				volume: -12,
			});
		});
```

**Expected behavior**
I expected the onload callback to be fired when both/all buffers have been loaded.

**What I've tried**
With just one item in the map, I verified that the `onload` callback fires. With more than one, it doesn't seem to work. In my personal code, I even replaced the audio file of the second player to be the same audio file as the first player, and the `onload` callback never fired.

```
  const audioSamples = new Tone.Players({
    sampleA: "assets/some_valid_url.m4a",
    sampleB: "assets/some_valid_url.m4a", // `onload` never called even if I use the same file!
  }, {
    onload: () => {
     // this line never seems to fire.
      console.log('loaded audio');
    },
    onerror: (e) => {
      console.log('players error ', e);
    },
  });
```

**Comment by ifjorissen** [comment]
Update:

This could be an issue with ToneAudioBuffers-- I don't see any test that confirms that `onload` is called when multiple urls are provided. (https://github.com/Tonejs/Tone.js/blob/master/Tone/core/context/ToneAudioBuffers.test.ts)

In the meantime, I'm going to `await Tone.ToneAudioBuffers.loaded()` since that seems pretty close to what I need right now.

**Comment by DarshanDixit05** [comment]
I want to work on this issue , please assign it to me.

**Comment by tambien** [comment]
There is a test that confirms that multiple audio files can be loaded: 

https://github.com/Tonejs/Tone.js/blob/6020401d9ef6bebd671f15865887869fac5bc80d/Tone/source/buffer/Players.test.ts#L36

**Comment by ifjorissen** [comment]
Yes I saw that test. There is no test that verifies that additional parameters are respected when multiple files are loaded.

**Comment by tambien** [comment]
Not following, locally i just switched it to this and it still worked: 

```ts

const players = new Players(
	{
		test0: "./test/audio/sine.wav",
		test1: "./test/audio/sine.wav",
	},
	{
		onload: () => {
			expect(players.player("test0")).to.be.instanceOf(
				Player
			);
			expect(players.player("test0").buffer.loaded).to.be
				.true;
			expect(players.player("test1")).to.be.instanceOf(
				Player
			);
			expect(players.player("test1").buffer.loaded).to.be
				.true;
			expect(players.loaded).to.be.true;
			players.dispose();
		},
	}
);

```

But maybe i'm missing something, if you're able to create a codesandbox that would help
